internal/controller/retry/retry.go

```diff
@@ -0,0 +1,33 @@
+package retry
+
+import (
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+)
+
+var RetryBackoffTwoMinutes = wait.Backoff{
+	Steps:    120,
+	Duration: time.Second,
+	Factor:   1.0,
+	Jitter:   0.1,
+}
+
+func isRetriable(err error) bool {
+	return apierrors.IsInternalError(err) || apierrors.IsServiceUnavailable(err) || net.IsConnectionRefused(err)
+}
+
+func RetryOnRetriable(backoff wait.Backoff, fn func() error) error {
+	return retry.OnError(backoff, isRetriable, fn) //nolint:wrapcheck
+}
+
+func isConflictOrRetriable(err error) bool {
+	return apierrors.IsConflict(err) || apierrors.IsInternalError(err) || apierrors.IsServiceUnavailable(err) || net.IsConnectionRefused(err)
+}
+
+func RetryOnConflictOrRetriable(backoff wait.Backoff, fn func() error) error {
```

Comment:
What's this for?

Reply:
This retry utility function will be used for updating the SiteConfig CR from the reconciler in a subsequent PR.